File: 프로그래머스/lv1/42862. 체육복/체육복.java
class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int answer = 0;
        int[] clothes = new int[n+2];
        for(int lostPersion : lost){
            clothes[lostPersion]--;
        }
        for(int reservedPerson : reserve){
            clothes[reservedPerson]++;
        }
        
        for(int i = 1; i< clothes.length-1; i++){
            if(clothes[i] == -1){
                if(clothes[i-1] == 1){
                    clothes[i-1]--;
                    answer++;
                }else if(clothes[i+1] == 1){
                    clothes[i+1]++;
                    answer++;
                }
            }else answer++;
        }
        return answer;
    }
}

File: 프로그래머스/lv1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    answer = 0
    arr = []
    for i in range(n+2):
        arr.append(0)

    for i in lost:
        arr[i] -= 1
    
    for i in reserve:
        arr[i] += 1 
    
    for i in range(1,n+1):
        if(arr[i] == -1):
            if(arr[i-1] == 1):
                arr[i-1] -= 1
                answer += 1
            elif(arr[i+1] == 1):
                answer += 1
                arr[i+1] -= 1
        else:
            answer+=1
        print(arr)
    return answer